Enable i2s_codec sample on mcxn5xx_evk

TomasBarakNXP opened the pull request:
- enable micfil on mcxn5xx_evk
- enable da7212 codec on mcxn5xx_evk

depends:
- [x] #96980 
- [x] #96978

VitekST committed:
soc: mcxnx4x: Instantiate MICFIL

Add the micfil node to nxp_mcxnx4x_common.dtsi.

Signed-off-by: Vit Stanicek <vit.stanicek@nxp.com>

TomasBarakNXP committed:
board: mcx_n5xx: Enable MICFIL and da7212 codec on mcx_n5xx

- in order to enable micfil on the mcx_n5xx board, it is necessary to
  enable the appropriate clock in the board.c
- create the proper pin_mux for micfil and da7212 codec and i2c which is
  used to communicate with the codec to enable micfil and da7212 codec
  in the mcx_n5xx dts

Signed-off-by: Tomas Barak <tomas.barak@nxp.com>

TomasBarakNXP committed:
board: mcx_nx4x: format board.c

- format mcx_nx4x_evk board.c which is not formatted according to the
  coding standards
- improve code readability and maintainability

Signed-off-by: Tomas Barak <tomas.barak@nxp.com>

TomasBarakNXP committed:
samples: i2s_codec: enable i2s_codec sample for mcx_n5xx_evk

- in order to test the loopback from PDM to I2S on the mcx_n5xx_evk
  it is necessary to add the dts nodes for SAI1 and MICFIL channels
- note that the configuration is using the newly introduced stream
  parameters (bytes_per_sample, sample_width, extra_blocks)
- add mcx_n5xx_evk/mcxn547/cpu0 to sample.yaml in order to allow this
  platform to be part of the CI tests

Signed-off-by: Tomas Barak <tomas.barak@nxp.com>

dbaluta commented on boards/nxp/mcx_nx4x_evk/board.c:
```diff
@@ -18,14 +18,16 @@
 #define BOARD_USB_PHY_TXCAL45DM (0x07U)
 
```
A commit message should be focused on *why* the changes is needed and less on *what* is does. What it does should be somehow obvious from the code (if it is not then explain it briefly in the commit message).

So here your commit message should be something like this:

```
 board: mcx_nx4x: Fix formatting 

In order to make style more consistent and code easier to read
fix some style issues:
- add enum values each on a single line
- fixes multiple whitespaces in macro definitions

Signed-off-by: Tomas Barak <tomas.barak@nxp.com>
```

If those were detected by a tool (e.g checkpatch.pl) please also mention this in the commit message.


Also, this change can be marked as trivial sent in a different PR and merged faster your if current PR is delayed by some dependencies.

dbaluta commented on samples/drivers/i2s/i2s_codec/boards/mcx_n5xx_evk_mcxn547_cpu0.conf:
```diff
@@ -0,0 +1,14 @@
+# Copyright 2025 NXP
+#
+# SPDX-License-Identifier: Apache-2.0
```
Commit message body is not giving us any extra info. We should try to use more of a natural language description of what the change is about focusing mostly on **why** the change is needed.

Also mention in the commit message things that are out of the ordinary which people reading the git log history might want to be aware about.

Commit title is nice and clear mostly saying what we need to now.

So, the commit message should look something like this:

```
samples: i2s_codec: enable i2s_codec sample for mcx_n5xx_evk

In order to test <insert your scenario here> on mcx_n5xx_evk we need to
configure dts nodes for SAI1 interface and MICFIL channels and
also enable some config options for DMA, I2S and DMIC.

Note that we are using newly introduced I2s params (bytes_per_sample, sample_widht, extra_blocks).

```
Signed-off-by: Tomas Barak <tomas.barak@nxp.com>

```

dbaluta commented on boards/nxp/mcx_nx4x_evk/board.c:
```diff
@@ -136,7 +136,7 @@ void board_early_init_hook(void)
 
 	CLOCK_SetupExtClocking(BOARD_XTAL0_CLK_HZ);
 
```
I wonder if you can enable them in separate patches? If one of them breaks the system one of them would have a hard time doing a bisection.

Please use 1 commit per a logical change so that code can be easier to be reviewed and also bisection and debugging in the future be easier.

ZhaoxiangJin commented on boards/nxp/mcx_nx4x_evk/board.c:
```diff
@@ -450,6 +451,15 @@ void board_early_init_hook(void)
+	CLOCK_AttachClk(kPLL1_CLK0_to_MICFILF);
+	CLOCK_EnableClock(kCLOCK_Micfil);
+
+	PORT0->PCR[16] = 0x00001900;
```
Hello @TomasBarakNXP, `PORT0->PCR[16] = 0x00001900;` and `PORT0->PCR[17] = 0x00001900;`  not needed now

ZhaoxiangJin commented on boards/nxp/mcx_nx4x_evk/board.c:
```diff
@@ -213,6 +213,7 @@ void board_early_init_hook(void)
 
 #if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(gpio0))
 	CLOCK_EnableClock(kCLOCK_Gpio0);
+	CLOCK_EnableClock(kCLOCK_Port0);
```
`CLOCK_EnableClock(kCLOCK_Port0);` is now not needed

TomasBarakNXP replied:
removed.

TomasBarakNXP committed:
soc: nxp: mcxn: remove trgmux dependency

- since trgmux IP is not present on mcxn5xx platform, remove
  the dependency to fix build issues

Signed-off-by: Tomas Barak <tomas.barak@nxp.com>

TomasBarakNXP commented:
@dleach02 please help with review and merge.

dleach02 commented on boards/nxp/mcx_nx4x_evk/board.c:
```diff
@@ -18,14 +18,16 @@
 
 /* Board xtal frequency in Hz */
-#define BOARD_XTAL0_CLK_HZ                        24000000U
+#define BOARD_XTAL0_CLK_HZ    24000000U
```
Not a blocking comment but in the future, separate the format changes in a standalone commit from any functional changes. 

For folks who walk through the commits to review them independently, it is easier on their time to see a commit that says it is just format changes with out any functional changes.
